File: backend/car-catalog-backend/src/main/java/com/wswork/carcatalog/repository/ModeloRepository.java
package com.wswork.carcatalog.repository;

import com.wswork.carcatalog.entity.Marca;
import com.wswork.carcatalog.entity.Modelo;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface ModeloRepository extends JpaRepository<Modelo, Long> {
    
    // Buscar modelos por marca
    List<Modelo> findByMarcaId(Long marcaId);
    
    // Buscar modelo por nome
    List<Modelo> findByNomeContainingIgnoreCase(String nome);
    
    // Buscar modelo por marca e nome
    List<Modelo> findByMarcaIdAndNomeContainingIgnoreCase(Long marcaId, String nome);
    
    // Buscar modelo por marca e nome exato
    Optional<Modelo> findByNomeAndMarca(String nome, Marca marca);
}

File: backend-kotlin/car-catalog-kotlin/bin/main/com/wswork/carcatalog/repository/ModeloRepository.kt
package com.wswork.carcatalog.kotlin.repository

import com.wswork.carcatalog.entity.Modelo
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ModeloRepository : JpaRepository<Modelo, Long> {
    fun findByMarcaId(marcaId: Long): List<Modelo>
    fun findByNome(nome: String): Modelo?
    fun existsByNome(nome: String): Boolean
}
